metrics/src/utils.rs

```diff
@@ -0,0 +1,41 @@
+use super::record_stats::Stats;
+use super::ErrorObj;
+use async_nats as nats;
+use serde::Serialize;
+use std::boxed::Box;
+use std::str::FromStr;
+
+#[derive(Debug, Clone)]
+pub struct NatsSubjectComponent(pub String);
+
+#[derive(Serialize)]
+pub struct DroneStatsMessage {
```

Comment:
I'm fine with this living here for now, but just flagging that we will eventually want to move it to `/core` so that we can use it in the controller for scheduling.